### DISCUSSION TITLE
My app with ANE crashes after iOS 14.1 upgrade

### DISCUSSION THREAD

**Discussion by fhassani** [airsdk/Adobe-Runtime-Support#621] [Q&A]
My app crashes everytime I call my custom ANE after upgrading to iOS 14.1.

The previous iOS versions are working fine.

I am using AIR SDK 32.0 and iPhoneOS13.1.sdk for compiling and packaging.

I removed my ANE and I tried packaging with 33.1.1.300 and I get this error:

Error occurred while packaging the application:

Undefined symbols for architecture armv7:

  "___darwin_check_fd_set_overflow", referenced from:

      __ZN26PlatformPrimitiveSocketTCP16HaveIncomingDataEv in libRuntimeHMAOT.arm-air.a(OSXPrimitiveSocket.o)

      __ZN14PlatformSelect12WaitForEventEjPb in libRuntimeHMAOT.arm-air.a(OSXPrimitiveSocket.o)

      __ZN20PlatformSecureSocket17WaitOnSocketErrorElNS_15WaitReadOrWriteEb in libRuntimeHMAOT.arm-air.a(CTSecureSocket.o)

      __ZN17MacDatagramSocket11ReceiveImplEPhPiR11FlashStringRiS3_S4_i in libRuntimeHMAOT.arm-air.a(OSXDatagramSocket.o)

      __ZN26PlatformPrimitiveSocketTCP16HaveIncomingDataEv in libRuntimeHMAOT.arm-air.a(OSXPrimitiveSocket.o)

      __ZN14PlatformSelect12WaitForEventEjPb in libRuntimeHMAOT.arm-air.a(OSXPrimitiveSocket.o)

      __ZN20PlatformSecureSocket17WaitOnSocketErrorElNS_15WaitReadOrWriteEb in libRuntimeHMAOT.arm-air.a(CTSecureSocket.o)

      __ZN17MacDatagramSocket11ReceiveImplEPhPiR11FlashStringRiS3_S4_i in libRuntimeHMAOT.arm-air.a(OSXDatagramSocket.o)

Then I tried to package using iPhoneOS14.0.sdk (dowloaded from https://resources.airnativeextensions.com/ios/), and I get this error:

ld: symbol(s) not found for architecture armv7

Please help!

**Comment by ajwfrost** [comment]
Can I just confirm:
"I am using AIR SDK 32.0 and iPhoneOS13.1.sdk for compiling and packaging"
AIR SDK v32 is likely to have an issue with this, you'd need to update to a recent 33.1 version..

**Comment by fhassani** [comment]
Thanks Andrew for your reply!

The combination AIR SDK 32.0 and iPhoneOS13.1.sdk is the one I am using for few months already. But after upgrading iPads to iOS 14.1, the application crashes. 

So I tried to use AIR SDK 33.1.1.300 with iPhoneOS13.1.sdk, but I cannot package. I got the error:

ld: symbol(s) not found for architecture armv7

**Comment by ajwfrost** [comment]
I would say that you can use 33.1.1.x up to (but not including) 300, with iPhoneOS13.1 SDK, but thenfrom 33.1.1.300 onwards it would need to be iPhoneOS14.x SDK..

But also, what is your MacOS version? If you're targeting 32-bit phones then you need to remain on Mojave...

**Comment by fhassani** [comment]
Do you think that AIR SDK 33.1.1.x combined with iPhoneOS13.1.sdk will solve the crash problem on iOS 14.1?

Do you know where I can download iPhoneOS14.1 SDK? I cannot find it here:  https://resources.airnativeextensions.com/ios/.

I use Windows to package the app and the ANE and MacOS Catalina to create the library for the ANE.

Thanks again!

**Comment by marchbold** [comment]
@fhassani I'm uploading it now (to that site above).

**Comment by marchbold** [comment]
@fhassani Is there any need for you to specify an iOS SDK? Normally you can just use the AIR SDK without it unless you need some of the features in the latest SDK.

**Comment by fhassani** [comment]
I tried without a long time ago and I had errors. I will try again this time.
Thank you!

**Comment by fhassani** [comment]
I tried with AIR SDK 33.1.1.259 without specifying any iOS SDK. I was able to package, but the app still crashes when I try to call my custom ANE on an iOS 14.1 iPad.

My ANE is packaged using AIR SDK 33.1.1.259.

When I tried to use iPhoneOS14.1 SDK, I have the packaging error: ld: symbol(s) not found for architecture armv7.

**Comment by ajwfrost** [comment]
Can I check how you built your ANE? was that using a specific iPhoneOS SDK?
And are you able to try with the 33.1.1.300 SDK which has the 14.1 definitions in it?
thanks

**Comment by fhassani** [comment]
Hi, 
My ANE was packaged using AIR SDK 33.1.1.259 whithout specifing any iPhoneOS SDK.
If I build the ANE with 33.1.1.300 SDK, should I build the App with 33.1.1.259?
If I build the ANE with 33.1.1.300 SDK, is my app going to work with previous versions of iOS?

**Comment by fhassani** [comment]
I built the ANE with 33.1.1.300 SDK, and the App with 33.1.1.259 whithout specifing any iPhoneOS SDK: when I tried to load to Apple, I got the following error: Please update iTMSTransporter to a newer version. (4107)

Then I built the ANE with 33.1.1.300 SDK, and the App with 33.1.1.300 whithout specifing any iPhoneOS SDK: when I tried to load to Apple, I got the same error: Please update iTMSTransporter to a newer version. (4107)

**Comment by fhassani** [comment]
I tried the 2 scenarios:
I built the ANE with 33.1.1.300 SDK, and the App with 33.1.1.259 whithout specifing any iPhoneOS SDK: it still crashes
I built the ANE with 33.1.1.300 SDK, and the App with 33.1.1.300 whithout specifing any iPhoneOS SDK: it still crashes

**Comment by marchbold** [comment]
Can you elaborate on what your ANE does or someway to reproduce the error?

**Comment by fhassani** [comment]
I am going to send you the ANE and its src code quickly ...

**Comment by fhassani** [comment]
Where can I send the ANE and src code?

**Comment by fhassani** [comment]
Hi, 
were you able to reproduce the error?
Do you need any information or action from me to solve this?
Thank you

**Comment by marchbold** [comment]
@fhassani Unfortunately I am extremely busy at the moment, feel free to post any snippets here and we'll help as we can. I suggest you try to isolate the part of your code that is crashing.

**Comment by fhassani** [comment]
@marchbold I have sent you by email the ANE and its code and how to call it.
The App crashes everytime the ANE is called.
Thank you
